plugins/calendar-action/src/index.ts

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright © 2020 Anticrm Platform Contributors.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import type { Ref, TxUpdateDoc } from '@anticrm/core'
+import core, { DeferredPromise } from '@anticrm/core'
+import type { CalendarService, Event } from '@anticrm/calendar'
+import { setResource } from '@anticrm/platform'
+import calendar from '@anticrm/calendar'
+import { Action } from '@anticrm/action'
+
+const wait = async <T>(p: Promise<T>, delay: number): Promise<null | T> => {
+  if (delay <= 0) {
+    return null
+  }
+
+  let tid: number | undefined
+  let delayResolve = (x: null): void => {}
+  const res = await Promise.race([
+    p,
+    new Promise<null>(resolve => {
+      delayResolve = resolve
+      tid = setTimeout(() => resolve(null), delay) as unknown as number
+    })
+  ])
+
+  if (res !== null) {
+    delayResolve(null)
+
+    if (tid !== undefined) {
+      clearTimeout(tid)
+    }
+  }
+
+  return res
+}
+
+const waitForEvent = new Action()
```

Comment:
So all of this is to check for event to be complete for a moment.

I personally think, better to have a boolean field for event named 'isComplete' and derived data to update this field on event updates.

So UI could allow to Move Application from one state to another state in case associated event is marked as isComplete. 

In this case Actions will not be required to be executed on server at all, and we will only need an UI action contribution to create an event.